src/Compilers/Core/CodeAnalysisTest/Diagnostics/CompilationWithAnalyzersTests.cs

```diff
@@ -67,5 +67,14 @@ public void GetAnalyzerTelemetry()
             var analyzerTelemetry = compWithAnalyzers.GetAnalyzerTelemetryInfoAsync(analyzer, CancellationToken.None).Result;
             Assert.Equal(0, analyzerTelemetry.SymbolActionsCount);
         }
+
+        [Fact]
+        public void TestIsDiagnosticAnalyzerSuppressedWithExceptionInSupportedDiagnostics()
+        {
+            // Verify IsDiagnosticAnalyzerSuppressed does not throw an exception when 'onAnalyzerException' is null.
+            var analyzer = new AnalyzerThatThrowsInSupportedDiagnostics();
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary);
+            _ = CompilationWithAnalyzers.IsDiagnosticAnalyzerSuppressed(analyzer, options, onAnalyzerException: null);
```

Comment:
@jcouv I added this unit test for null `onAnalyzerException` value passed down to the analyzer executor, but I could not get the test to fail even on master. NRE seems to be silently swallowed below: https://github.com/dotnet/roslyn/blob/63506f6714a2b0232367cc54f07181c284a69a3b/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs#L1582-L1589

At least now we ensure no NRE is thrown in the above code by the analyzer executor for this test.

Reply:
Makes sense. Thanks!

---
In reply to: [468924831](https://github.com/dotnet/roslyn/pull/46409#discussion_r468924831) [](ancestors = 468924831)